smart-workflow-test/src_test/com/axonivy/utils/smart/workflow/model/azureopenai/TestVariableUtils.java

```diff
@@ -0,0 +1,100 @@
+package com.axonivy.utils.smart.workflow.model.azureopenai;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.util.List;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import com.axonivy.utils.smart.workflow.azureopenai.AzureAiDeployment;
+import com.axonivy.utils.smart.workflow.azureopenai.utlis.VariableUtils;
+
+import ch.ivyteam.ivy.environment.AppFixture;
+import ch.ivyteam.ivy.environment.IvyTest;
+
+@IvyTest
+public class TestVariableUtils {
+  private static final String DEPLOYMENTS_PREFIX = "AI.Providers.AzureOpenAI.Deployments";
+  private static final String TEST_DEPLOYMENT_1 = "test-gpt-4-1-mini";
+  private static final String TEST_DEPLOYMENT_2 = "test-o4-mini";
+
+  private static final String MODEL_1 = "gpt-4.1-mini";
+  private static final String MODEL_2 = "o4-mini";
+
+  private static final String API_KEY_1 = "${decrypt:test-key-1}";
+  private static final String DECRYPTED_API_KEY_1 = "test-key-1";
+  private static final String API_KEY_2 = "${decrypt:test-key-2}";
+
+  @BeforeEach
+  void setup(AppFixture fixture) {
+    // Setup test deployments with Model and APIKey fields
+    fixture.var(DEPLOYMENTS_PREFIX + "." + TEST_DEPLOYMENT_1 + ".Model", MODEL_1);
+    fixture.var(DEPLOYMENTS_PREFIX + "." + TEST_DEPLOYMENT_1 + ".APIKey", API_KEY_1);
+
+    fixture.var(DEPLOYMENTS_PREFIX + "." + TEST_DEPLOYMENT_2 + ".Model", MODEL_2);
+    fixture.var(DEPLOYMENTS_PREFIX + "." + TEST_DEPLOYMENT_2 + ".APIKey", API_KEY_2);
+  }
+
+  @Test
+  void testGetAllDeployments() {
+    List<AzureAiDeployment> deployments = VariableUtils.getDeployments();
+
+    assertThat(deployments).isNotNull();
+    assertThat(deployments).hasSize(2);
+
+    List<String> deploymentNames = deployments.stream().map(AzureAiDeployment::getName).toList();
+
+    assertThat(deploymentNames).containsExactlyInAnyOrder(TEST_DEPLOYMENT_1, TEST_DEPLOYMENT_2);
+  }
+
+  @Test
+  void testContainCorrectModelAndApiKey() {
+    List<AzureAiDeployment> deployments = VariableUtils.getDeployments();
+
+    AzureAiDeployment deployment1 = deployments.stream().filter(d -> TEST_DEPLOYMENT_1.equals(d.getName())).findFirst()
+        .orElse(null);
+
+    assertThat(deployment1).isNotNull();
+    assertThat(deployment1.getName()).isEqualTo(TEST_DEPLOYMENT_1);
+    assertThat(deployment1.getModel()).isEqualTo(MODEL_1);
+    assertThat(deployment1.getApiKey()).isEqualTo(DECRYPTED_API_KEY_1);
+  }
+
+  @Test
+  void testNonExistentDeployment() {
+    AzureAiDeployment deployment = VariableUtils.getDeploymentByName("non-existent-deployment");
+
+    assertThat(deployment).isNull();
+  }
+
+  @Test
+  void testReturnNullForBlankName() {
```

Comment:
a minor, here by example, but applies to all new tests.
 I'd omit the 'test' prefixes ... I think it's Junit4 stile and doesn't add any meaning to the thing under test. -> simple "returnNullForBlankName" would be easier to read.